addon/services/firestore.js

```diff
@@ -0,0 +1,15 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+
+export default Service.extend({
+  firebase: service(),
+  instance: computed('firebase', function() {
```

Comment:
Really appreciate for your time on taking a stab at this.

I'm not really liking the `firestore.instance` but I could see why you did it this way. I could probably refactor in the future should there be a cleaner solution that we can omit the `.instance` part. For now, this will do.

My only suggestion is maybe we can make this as not a computed property. We can assign the `instance` property in the `init()` hook. The `settings` will also be a standalone property so that other devs can override the service and provide their own settings if needed.

e.g.


```javascript
export default Service.extend({
  ...

  settings: { timestampInSnapshots: true },

  init(...args) {
    this._super(...args);

    const firestore = this.get('firebase').firestore();

    firestore.settings(this.get('settings'));

    this.set('instance', firestore);
  }
});
```

Reply:
I agree on both items:
- as a service needs to be an extension of Service, I could not see a way around this (the `instance` part for now.
- I changed the code so the instance gets set on init (and settings are overridable)

I pushed a new commit.

Changes I think could be added but should not prevent this PR from merging:
-> get settings from ENV
-> Update your mock-cloud-firestore so it implements a `settings` method (tests were failing because of that, so I added the if-guard)